### DISCUSSION TITLE
Proposal: winget release

### DISCUSSION THREAD

**Discussion by raphael-zenhaeusern-legic** [git-for-windows/git#4399] [Ideas]
### Abstract
Integrating automatic update of the winget repository manifest as part of the release pipeline would add good value for both businesses and individuals using the package manager.

### Why winget
[Windows Package Manager](https://github.com/microsoft/winget-cli) (or winget) is Microsofts own package manager for Windows allowing for installation of apps form the Microsoft store as well as other sources in various installer formats. It now is even somewhat integrated with Microsoft Intune (device and software management for businesses) which leads to some further adoption.

### Current state
Git for Windows is already available on winget but the entry in the repository seems to be maintained manually by individuals outside this project. Due to this, the "current" version in the repository might be outdated. We learned that in the last days when we wanted to immediately update to the latest release due to the CVEs involved.

### Proposed change
The [winget repository](https://github.com/microsoft/winget-pkgs) offers software and scripts to facilitate creation and submission of manifest files as part of a release pipeline. Integrating this with your pipeline should be reasonable low effort and - in contrast to a Microsoft Store release - requires no change to the packaging of the software.

**Comment by rimrul** [comment]
> Git for Windows is already available on winget but the entry in the repository seems to be maintained manually by individuals outside this project. 

Yes, that is correct. We are currently not involved in any package managers (winget, scoop, chocolatey, ...).

> The [winget repository](https://github.com/microsoft/winget-pkgs) offers software and scripts to facilitate creation and submission of manifest files as part of a release pipeline. Integrating this with your pipeline should be reasonable low effort and - in contrast to a Microsoft Store release - requires no change to the packaging of the software.

We'd probably want to generate the package manifests in a separate workflow that is triggered on successful completion of the [`release-git` workflow](https://github.com/git-for-windows/git-for-windows-automation/blob/main/.github/workflows/release-git.yml). We'd then need to manually open the corresponding PRs (to allow for potentially requested changes).

Looking at https://github.com/microsoft/winget-pkgs/pull/99422 it also looks like they'd expect us to commit to providing an unattended install, which I'm not sure we can do.

**Reply by raphael-zenhaeusern-legic** [reply]
The issue mentioned in https://github.com/microsoft/winget-pkgs/pull/99422 is specific to the warning popup with the 32bit installer. 64bit currently works fine with an unattended install. As you will phase out 32bit anyway, you could just stick with 64bit only as is currently done by the other contributors maintaining the repository entry.

**Reply by byjrack** [reply]
Something to have on the radar for any official manifest for git for windows is the scoping.  Currently the user scoped install still prefers Program Files and a UAC, but I was hoping to stay in the User's Program/Packages which is not an option in the manifest today.    The portable can work for the need, but that's not a winget option.

I was able to fake it a bit using --override 

`winget install --id Git.Git --override "/DIR=%LOCALAPPDATA%\Git /CURLOption=WinSSL /VERYSILENT"`

I haven't tried to fork the manifest yet as I wasn't sure if it would work at all.  Was going to toy with the InstallLocation schema item to see if that was respected, supports variable expansion, and passed into the IS installer.  Won't be able to escape the --override if you want to do any custom params that might have come from the ISS file, but that is what it is there for.

Not sure why the community manifest makes machine match user as it definitely through me off originally.

**Reply by rimrul** [reply]
> The issue mentioned in [microsoft/winget-pkgs#99422](https://github.com/microsoft/winget-pkgs/pull/99422) is specific to the warning popup with the 32bit installer. 

Yes, but it gives us a preview for a broader more general expectation we'd have to meet. That is not the only `SuppeessibleMsgBox` in the installer that could stop an unattended `/VERYSILENT /SUPRESSMSGBOXES` install.

https://github.com/git-for-windows/build-extra/blob/6dc2d669d487578f75219db84e04b27293cad04a/installer/install.iss#L1123

https://github.com/git-for-windows/build-extra/blob/6dc2d669d487578f75219db84e04b27293cad04a/installer/install.iss#L1131

https://github.com/git-for-windows/build-extra/blob/6dc2d669d487578f75219db84e04b27293cad04a/installer/install.iss#L2579


> 64bit currently works fine with an unattended install. 

So does 32bit, on 32bit (and ARM64) Windows. WinGet just happens to test all their packages on AMD64 machines, because those are readily available through GitHub actions hosted runners.

> As you will phase out 32bit anyway, you could just stick with 64bit only as is currently done by the other contributors maintaining the repository entry.

We could do that. We'll probably have to do either that or implement an option to override that `SupressibleMsgBox` if we want to provide WinGet Manifests, as we'd always fail the tests for the 32bit version.

**Comment by soredake** [comment]
https://github.com/git-for-windows/git/issues/4758
